### ISSUE TITLE
After an image has been uploaded, the user is unable to see it (only RV)

### ISSUE THREAD

**Issue by mrivoal** [dhlab-basel/Knora#1003]
Using Salsah 1.5, after a user has uploaded an image (i.e. `anything.admin@example.org` creating a `anything:Dingbild`), he is unable to view it properly afterwards. 

The image is only available to him in restricted view. 

It looks like there is a bug in permissions that didn't exist before.

**Comment by mrivoal** [comment]
And FYI, I am currently using your docker stack with the release v2.0.0. 
@loicjaouen says that I am in "developer mode".

**Comment by subotic** [comment]
Thanks for reporting it. I will have to take a closer look. I don't remember changing anything that has to do with permissions in the last few months.

**Comment by mrivoal** [comment]
Hello @subotic. I guess you haven't found the time to have a look at this issue... So this is just a kind reminder :)
In fact, we are going productive for some projects using Knora v2.1.0. and Salsah 1.5 and this issue will become a problem (quite) soon.

**Comment by subotic** [comment]
I've been able to replicate the error. Now I'm trying out to figure out the problem.

**Comment by mrivoal** [comment]
> So I suppose that with Salsah 1.5 you mean the version of Salsah1 (non-angular) which was released at the time Knora v2.1.0 was released?

Yes! Sorry for the late answer.
And thanks for having a look at it.

**Comment by subotic** [comment]
`Salsah1` is receiving at login (with `/Knora_login`) a cookie with an authentication token, but does not send it when accessing images so `Sipi` cannot authenticate the client and thus always returns the lowest resolution.

**Comment by benjamingeer** [comment]
Soon I'm going to change Knora to use JWT tokens to authorise image access (which will also work with SALSAH 1), so we won't need cookies for that anymore.

**Comment by subotic** [comment]
I suspect that the behavior of the `img` tag in browsers regarding cookies has changed and that there is some additional configuration needed.

> Soon I'm going to change Knora to use JWT tokens

What is the timeline on this? If it is going to happen next week, then I won't spend too much time on this issue.

Or can I implement a quick workaround with tokens, by appending the token to each image URL? You could then make it more sophisticated without any time pressure. There are some additional gotchas which we discussed in #520.

**Comment by benjamingeer** [comment]
> What is the timeline on this?

I was thinking of doing this in January.

> Or can I implement a quick workaround with tokens, by appending the token to each image URL?

That's what I was going to do. :) But I had completely forgotten about the gotchas in #520.

**Comment by subotic** [comment]
Ok, then I will investigate the img-cookie problem a bit more. If I don't find a solution, I will add JWT tokens to `img-urls`, so that Salsah1 is working again. Then in January, we can implement a more complete solution that will also satisfy #520.

**Comment by gfoo** [comment]
and for a demo purpose, as workaround, is there a way to disable Sipi permissions interaction with Knora ? (reminds me something but cannot find it anymore in the doc...)

**Comment by subotic** [comment]
yes, of course. This is what BEOL is doing. In the preflight script always return `allow`

**Comment by subotic** [comment]
```
if prefix == "tmp" then
        -- always allow access to tmp folder
        return 'allow', filepath
end


if prefix == "knora" then
        -- always allow access to any Knora imager
        return 'allow', filepath
end
```

**Comment by gfoo** [comment]
@loicjaouen  nice to have in our demo stack :)

**Comment by subotic** [comment]
I think I have found the problem. It was a configuration issue when running the stack locally. Now, when using the new way of starting the stack `$sbt dockerComposeUp` (from the root of the repository) and opening Salsah1 in the browser through `http://localhost:3335`, it should work again with Sipi images.

I have found another bug #1086, which would prevent showing the preview image after uploading it in the create resource window.  The preview image URL was hardcoded, which is now configurable (see `docker-compose.yml`)
